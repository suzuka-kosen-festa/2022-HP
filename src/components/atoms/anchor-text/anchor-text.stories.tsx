import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { AnchorText } from ".";

type T = typeof AnchorText;
type Story = ComponentStoryObj<T>;

const data = {
  link: "#",
  role: "link",
  sentence: "This is test",
  tabindex: 0,
};

export default {
  args: { children: data.sentence, href: data.link, role: data.role, tabIndex: data.tabindex },
  component: AnchorText,
} as ComponentMeta<T>;

export const Default: Story = {};
