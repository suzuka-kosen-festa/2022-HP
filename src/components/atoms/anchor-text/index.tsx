import tw from "twin.macro";

const AnchorText = tw.a`font-zen text-xs sm:text-text text-base`;

export { AnchorText };
