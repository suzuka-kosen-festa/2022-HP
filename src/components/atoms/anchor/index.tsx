import tw from "twin.macro";

const AnchorText = tw.a`font-zen text-white text-base`;

export { AnchorText }
