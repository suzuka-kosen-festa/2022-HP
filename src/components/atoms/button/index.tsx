import React from "react";
import tw from "twin.macro";
import { ButtonProperties } from "../../../models";
import { Text } from "../text";

const ButtonBox = tw.button`bg-white hover:bg-button-hover flex flex-row items-start px-8 py-2 gap-2.5 rounded-[2rem]`;

const Button = <T extends AnyComponent>({
  children,
  boxStyles,
  textStyles,
  as,
  ...rest
}: ButtonProperties<T>): JSX.Element => (
  <ButtonBox css={boxStyles} as={as} {...rest}>
    <Text css={[tw`text-xs font-bold text-black`, textStyles]}>{children}</Text>
  </ButtonBox>
);

export { Button };
