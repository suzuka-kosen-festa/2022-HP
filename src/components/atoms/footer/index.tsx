import React from "react";
import tw from "twin.macro";
import { Text } from "../text";
const Footer: React.FC = () => (
  <footer css={tw` w-screen`}>
    <Text css={tw`bg-black text-center p-2`}>&copy; 第57回鈴鹿高専祭実行委員会</Text>
  </footer>
);

export { Footer };
