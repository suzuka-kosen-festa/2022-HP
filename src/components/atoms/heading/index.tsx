import type { FC } from "react";
import tw, { css } from "twin.macro";
import type { TwStyle } from "twin.macro";
import type { HeadingProperties, ColorVariant } from "./type/model";

const HeadingBase = tw.h1`font-zen text-text text-5xl select-none`;

// eslint-disable-next-line consistent-return
const getColor = (color: ColorVariant): TwStyle => {
  switch (color) {
    case "yellow":
      return tw`text-primary-yellow`;
    case "green":
      return tw`text-primary-green`;
    case "blue":
      return tw`text-primary-blue`;
    case "purple":
      return tw`text-primary-purple`;
    case "red":
      return tw`text-primary-red`;
    // no default
  }
};

const Heading: FC<HeadingProperties> = ({ color = "yellow", children, ...rest }) => (
  <HeadingBase
    css={css`
      &::first-letter {
        ${getColor(color)}
      }
    `}
    {...rest}
  >
    {children}
  </HeadingBase>
);

export { Heading };
