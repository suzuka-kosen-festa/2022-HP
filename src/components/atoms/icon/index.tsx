import { InlineIcon } from "@iconify/react";
import { memo } from "react";
import tw from "twin.macro";
import type { IconProperties } from "./type/model";

const IconBox = tw.div`flex items-center justify-center flex-shrink-0 w-6 h-6 sm:(w-12 h-12) min-h-full`;

const Icon = <T extends AnyComponent>({ icon, iconStyles, as, ...rest }: IconProperties<T>): JSX.Element => (
  <IconBox as={as} {...rest}>
    <InlineIcon icon={icon} css={[tw`w-full h-full`, iconStyles]} />
  </IconBox>
);

export default memo(Icon);
