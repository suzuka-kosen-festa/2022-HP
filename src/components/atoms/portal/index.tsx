import type { FC } from "react";
import { useMemo } from "react";
import { createPortal } from "react-dom";
import type { PortalProperties } from "./types/model";

const Portal: FC<PortalProperties> = ({ children, portalId }) => {
  const portalElement = useMemo<HTMLElement | null>(() => document.querySelector(portalId), [portalId]);
  if (!portalElement) {
    throw new Error(`Undefined Portal Id : ${portalId}`);
  }
  return createPortal(children, portalElement);
};

export { Portal };
