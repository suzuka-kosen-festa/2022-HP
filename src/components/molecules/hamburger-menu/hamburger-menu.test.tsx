import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { InlineIcon } from "@iconify/react";
import tw from "twin.macro";
import { Link } from "wouter";
import { AnchorText } from "../../atoms/anchor-text";

const headerLinks = [
  {
    link: "/",
    color: tw`text-primary-yellow`,
    name: "Top",
  },
  {
    link: "/member",
    color: tw`text-primary-red`,
    name: "Members",
  },
  {
    link: "/timetable",
    color: tw`text-primary-green`,
    name: "Schedule",
  },
  {
    link: "/bazaar",
    color: tw`text-primary-purple`,
    name: "Bazaar",
  },
  {
    link: "/sponsors",
    color: tw`text-primary-blue`,
    name: "Sponsors",
  },
];

// Mocking because menu can't open in jsdom
const Mock = () => (
  <div css={tw`absolute inset-y-0 right-4 inline-block md:hidden`}>
    <button
      css={tw`p-0.5 w-4 h-4 min-w-min min-h-full flex justify-center items-center`}
      type="button"
      aria-label="menu button"
      aria-haspopup="menu"
      aria-expanded
      aria-controls="hamburger-list"
      onClick={vi.fn()}
    >
      <div css={tw`flex items-center justify-center flex-shrink-0 w-4 h-4 min-h-full`}>
        <InlineIcon icon="eva:close-fill" css={tw`w-full h-full text-black text-2xl`} />
      </div>
    </button>
    <div css={tw`absolute top-0 left-0`}>
      <ul
        css={tw`flex flex-col justify-center items-center px-4 py-2 rounded-lg bg-white max-w-min shadow-xl space-y-1 z-10`}
        role="menu"
        aria-orientation="vertical"
        id="hamburger-list"
      >
        {headerLinks.map(headerLink => (
          <li role="menuitem" key={headerLink.name}>
            <Link to={headerLink.link}>
              <AnchorText css={[tw`hover:underline`, headerLink.color]}>{headerLink.name}</AnchorText>
            </Link>
          </li>
        ))}
      </ul>
    </div>
  </div>
);

describe("(components) molecules/hamburger-menu", () => {
  test("to be organisms", () => {
    const { container } = render(<Mock />);
    expect(container).toBeMolecule();
  });
  test("take snap shot", () => {
    const { container } = render(<Mock />);
    expect(container).toMatchSnapshot();
  });
});
