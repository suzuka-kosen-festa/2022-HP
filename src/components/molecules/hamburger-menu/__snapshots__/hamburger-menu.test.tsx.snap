// Vitest Snapshot v1

exports[`(components) molecules/hamburger-menu > take snap shot 1`] = `
<div>
  <div
    class="css-1iqlfdf-Mock"
  >
    <button
      aria-controls="hamburger-list"
      aria-expanded="true"
      aria-haspopup="menu"
      aria-label="menu button"
      class="css-1pclbbb-Mock"
      type="button"
    >
      <div
        class="css-15cs8in-Mock"
      >
        <span />
      </div>
    </button>
    <div
      class="css-3xglg4-Mock"
    >
      <ul
        aria-orientation="vertical"
        class="css-13737nk-Mock"
        id="hamburger-list"
        role="menu"
      >
        <li
          role="menuitem"
        >
          <a
            class="css-1vd7en6-Mock"
            href="/"
          >
            Top
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            class="css-1ea5aff-Mock"
            href="/member"
          >
            Members
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            class="css-xddjtl-Mock"
            href="/timetable"
          >
            Schedule
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            class="css-16racjb-Mock"
            href="/bazaar?select=eating"
          >
            Bazaar
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            class="css-dgw1rc-Mock"
            href="/sponsors"
          >
            Sponsors
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
